File: app/src/main/java/com/andreacioccarelli/billingprotectorsample/DetectionActivity.java
package com.andreacioccarelli.billingprotectorsample;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.app.Activity;
import android.os.Vibrator;
import android.support.design.widget.FloatingActionButton;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.andreacioccarelli.billingprotector.*;
import java.util.concurrent.TimeUnit;

public class DetectionActivity extends Activity {

    BillingProtector bp;
    private int i = 0;

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity);

        bp = new BillingProtector(this, false);
        runRefresh();

        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Vibrator vib = (Vibrator) getSystemService(VIBRATOR_SERVICE);
                vib.vibrate(100);

                runRefresh();
            }
        });
    }

    void runRefresh() {
        new Thread(new Runnable() {
            @Override
            public void run() {
                long mills = System.currentTimeMillis();
                updateData();

                long millsAfter = System.currentTimeMillis();
                long diff = millsAfter - mills;
                final long seconds = TimeUnit.MILLISECONDS.toSeconds(diff);

                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        Toast.makeText(DetectionActivity.this, "Re-computed. time: " + seconds + "s", Toast.LENGTH_SHORT).show();
                    }
                });
            }
        }).run();
    }

    @SuppressLint("SetTextI18n")
    void updateData() {
        final TextView mxp = findViewById(R.id.mxp);
        mxp.setText(
                "i=" + i++ +
                "isRootInstalled: " + bp.isRootInstalled() +
                "\n\npirateAppsList: " + bp.getPirateAppsList()
        );
    }
}

File: app/src/main/java/com/andreacioccarelli/billingprotectorsample/DetectionActivity.kt
package com.andreacioccarelli.billingprotectorsample

import android.annotation.SuppressLint
import android.os.Bundle
import android.app.Activity
import android.content.Context
import android.os.Vibrator
import android.support.design.widget.FloatingActionButton
import android.widget.TextView
import android.widget.Toast

import com.andreacioccarelli.billingprotector.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import java.util.concurrent.TimeUnit
import kotlin.coroutines.CoroutineContext
import kotlin.system.measureTimeMillis

class DetectionActivity : Activity() {

    private lateinit var bp: BillingProtector

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity)

        bp = BillingProtector(this)
        runRefresh()

        val fab = findViewById<FloatingActionButton>(R.id.fab)
        fab.setOnClickListener {
            val vib = getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
            vib.vibrate(100)

            runRefresh()
        }
    }

    private fun runRefresh() = CoroutineScope(Dispatchers.Main).launch {
        val millis = measureTimeMillis {  updateData() }
        val seconds = TimeUnit.MILLISECONDS.toSeconds(millis)

        runOnUiThread {
            Toast.makeText(this@DetectionActivity, "Re-computed. time: ${seconds}s", Toast.LENGTH_SHORT).show()
        }
    }

    private var i = 0
    @SuppressLint("SetTextI18n")
    private suspend fun updateData() {
        val mxp = findViewById<TextView>(R.id.mxp)
        mxp.text = "i=${i++}\nisRootInstalled: ${bp.isRootInstalled()}\npirateAppsList: ${bp.getPirateAppsListAsync()}"
    }
}
